backend/pkg/database/gorm/database.go

```diff
@@ -132,8 +132,8 @@ func initDataBase(config types.DBConfig) (*gorm.DB, error) {
 	//	SELECT COUNT(*) FROM findings WHERE ((findings.Data->'$.findingInfo.objectType' = JSON_QUOTE('Vulnerability') AND datetime(findings.Data->>'$.foundOn') <= datetime('2023-06-11T14:24:28Z')) AND (findings.Data->'$.invalidatedOn' is NULL OR datetime(findings.Data->>'$.invalidatedOn') > datetime('2023-06-11T14:24:28Z')))
 	idb = db.Exec(fmt.Sprintf("CREATE INDEX IF NOT EXISTS findings_by_type_and_foundOn_and_invalidatedOn_idx ON findings((%s), (%s), (%s), (%s))",
 		SQLVariant.JSONExtract("Data", "$.findingInfo.objectType"),
-		SQLVariant.CastToDateTime(SQLVariant.JSONExtractText("Data", "$.foundOn")),
-		SQLVariant.CastToDateTime(SQLVariant.JSONExtractText("Data", "$.invalidatedOn")),
+		SQLVariant.JSONExtractText("Data", "$.foundOn"),
+		SQLVariant.JSONExtractText("Data", "$.invalidatedOn"),
```

Comment:
I think that while not doing date-time casting the widget query will return to be slow - did you checked it?

Reply:
I need to double check, I'm going to deploy a version of this fix with SQLite enabled and test to see if the index gets used

Reply:
Double checked, and SQLite is using the index and it appears to be performing well without the datetime cast in the index itself.

This index might actually be better from a re-use perspective if we want to do a different comparison against the date fields because the index isn't specific about it needing to be a date too.